labs/b5.md

```diff
@@ -0,0 +1,364 @@
+![alt text][DHCP1]
+
+[DHCP1]: https://i.imgur.com/Muw9SFf.png "DHCP through Mac GUI"
+
+![alt text][DHCP2]
+
+[DHCP2]: https://i.imgur.com/g5ubmUO.png "DHCP through terminal"
+
+Notice anything special about the IP address in the above examples? [Hint: Look at the network block][private] Additionally, do you recall what must be done so that your unique IP address can communicate over the internet?
+
+To dive a bit deeper into DHCP, this diagram details the typical procedure for a client requesting an IP.
+![alt text][DHCP3]
+
+[DHCP3]: https://upload.wikimedia.org/wikipedia/commons/thumb/e/e4/DHCP_session.svg/260px-DHCP_session.svg.png "DHCP request steps"
+
+The client broadcasts a DHCPDISCOVER message essentially communicating that it wants an IP address. A DHCP server responds to the DHCPDISCOVER with a DHCPOFFER which contains an IP address for the client. If the client accepts the offer, it sends a DHCPREQUEST back to the server. The server acknowledges this by sending a DHCPACK back. The IP address is then considered leased to the machine and is valid for some period of time specified by the DHCP server. Once an IP address lease expires, the client must acquire a new IP address though it does have the option to renew a previous lease for the same IP address.
+
+## TCP and UDP
+Now we will transition into a discussion on the protocols at the transport layer. The two most well known protocols at this layer are Transmission Control Protocol (TCP) and User Datagram Protocol (UDP).
+
+TCP is a stateful stream oriented protocol that ensures reliable transport. TCP also has mechanisms to guarantee that information arrives intact and in order at the destination. 
+
+A TCP connection begins with something known as the TCP handshake. 
+![alt text][handshake]
+
+[handshake]: https://media.licdn.com/mpr/mpr/shrinknp_400_400/AAEAAQAAAAAAAAQfAAAAJDZjYTU5NDQ0LWQ0OTMtNDAxMS1hMDU1LTMzZjE5ZWM0MjkxYg.gif "TCP handshake"
+
+The TCP handshake consists of setting certain flags in the TCP header of packets exchanged between sender and receiver.  The sender initiating a TCP connection by first sending a SYN, a packet with the SYN flag set.  The server acknowledges this connection request by sending back a SYN-ACK, a packet with both the SYN and ACK flags set. The client acknowledges this by sending one final ACK back to the server, and the connection is then established.
+
+TCP then begins transmitting data and if it successfully arrives on the other end of the connection then an ACK is issued. Therefore if data is lost, reordered, or corrupted, TCP is capable of recognizing this and sends a request for retransmission of any lost data. 
+
+TCP also has a procedure to close connections. We only consider a graceful termination here, abrupt terminations have a different procedure we will not go over. Let's assume machine A wants to close its connection to machine B.
```

Comment:
I'd link them to more information if we're going to say it's beyond the scope of the topic

Reply:
added link